client/src/store/user/reducer/computation-functions.js

```diff
@@ -1,11 +1,54 @@
 export function putUserData(state, payload) {
-  console.log({
+
+  return {
     ...state,
-    ...payload
-  });
+    data:{
+      ...state.data,
+      ...payload
+    }
+  };
+}
 
+export function putUserToken(state, payload){
+  
   return {
     ...state,
-    ...payload
+    ...payload  
   };
```

Comment:
ovo mi ne djeluje dobro (to jest okej je, radice, ali radis nepotreban korak), ono sto predpostavljam da radis kad saljes ovaj payload, saljes ga {token: "neki string" }, ako je to slucaj onda ispravi da saljes samo 'neki string' , a onda ovdje kazes 
```
return {
  ...state,
  token: payload
}
```

a mozes i taj parametar drugi preimenovati u `token` i onda ces imati:
```
{
   ...state,
   token
}
```